src/VisualStudio/Core/Def/DocumentOutline/DocumentSymbolData.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    using SymbolKind = LanguageServer.Protocol.SymbolKind;
+
+    internal sealed class DocumentSymbolData
+    {
+        public string Name { get; }
+        public SymbolKind SymbolKind { get; }
+        public SnapshotSpan RangeSpan { get; }
+        public SnapshotSpan SelectionRangeSpan { get; }
+        public ImmutableArray<DocumentSymbolData> Children { get; }
+
+        public DocumentSymbolData(DocumentSymbol documentSymbol, ITextSnapshot originalSnapshot, ImmutableArray<DocumentSymbolData> children)
```

Comment:
nit: keep this type *pure* data (don't even have hte logic about translating spans here).  So have this take in all the prebaked data it needs.  the helper htat calls this can do that translation first.